fix: weapons, reverting some changes from XML pr

dudantas opened the pull request:
Reverting (and revising) some changes from: https://github.com/opentibiabr/canary/pull/740

Fixes using weapons event (onUseWeapon)
Fixes monsters spells invert order of target and caster

dudantas committed:
Fix weapons attack area

dudantas committed:
Fix

murilo09 committed:
Fix unscripted weapons and spells

dudantas committed:
Fix monsters spells targeting wrong

DeCarvalhoJr reviewed (commented):
Tested in some scenarios w/o any error
i was curious at first if it would fix diamond arrow (yes, it did) and then went to test other thinsg..

dudantas commented:
> Tested in some scenarios w/o any error i was curious at first if it would fix diamond arrow (yes, it did) and then went to test other thinsg..

There is still a bug, the spells of monsters in the area are in trouble. I'm investigating the cause.

DeCarvalhoJr commented:
> 

Weird cuz i made some tests with gaz'haragoth and abyssador (they both have aoe) and i got no errors..
unless its occurring with specific monsters.. could you list some names so i could make some tests here?

dudantas commented:
> Weird cuz i made some tests with gaz'haragoth and abyssador (they both have aoe) and i got no errors..
> unless its occurring with specific monsters.. could you list some names so i could make some tests here?

Demon, energy wave for example.

dudantas committed:
Some sonar fixes

Moved combat* from Spell class to shared_ptr (related to combatIdMap from LuaEnvironment)
Removed CombatSpell destructor

dudantas committed:
Sonar fixes

dudantas committed:
fix: migrate "new" pointers to std::shared_ptr

dudantas committed:
Merge branch 'main' into fix-monsters-spells

dudantas committed:
Fix some sonar reports
